module/zfs/mmp.c

```diff
@@ -158,6 +215,12 @@ mmp_thread(dsl_pool_t *dp)
 		    &mmp->mmp_thread_lock, start + mmp->mmp_interval,
```

Comment:
(defect) This doesn't try to "catch up" if the thread took longer to complete the last update than expected, so it is sleeping `mmp_interval` after the last write _completed_ rather than after the last write _started_.  It should only be sleeping `if (mmp->mmp_last_update + mmp->mmp_interval - get_hrtime() > 0)` (i.e. not at all if we have already passed the last interval) so that if the system is overloaded it will be writing MMP blocks as fast as it possibly can.